### PR TITLE
optick/1.4.0.0: new recipe

### PR DESCRIPTION
**Pull request by luizfeldmann**
Specify library name and version:  **optick/1.4.0.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/optick/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import is_msvc_static_runtime, is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=2.0.0"
```
Comment (uilianries) [inline recipes/optick/all/conanfile.py:9]:
```suggestion
required_conan_version = ">=1.54.0"
```

Please, do lock for Conan 2.x. The Conan 1.x is still mandatory for any PR: https://github.com/conan-io/conan-center-index/discussions/19104

File: recipes/optick/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        pass
+
+    def configure(self):
+        pass
+
```
Comment (uilianries) [inline recipes/optick/all/conanfile.py:35]:
```suggestion
```

Not really required, as its a shared-only library and there are not patches to be applied.

File: recipes/optick/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    name = "optick"
+    description = "Optick is a super-lightweight C++ profiler for Games"
+    license = "MIT"
+    url = "https://github.com/bombomby/optick"
+    homepage = "https://github.com/bombomby/optick"
+    topics = ("profiler", "games")
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {}
+    default_options = {}
```
Comment (uilianries) [inline recipes/optick/all/conanfile.py:21]:
```suggestion
```

No not need placeholders, Conan client is using the same values already and will not raise warnings. Please, try to keep it simpler as possible.

File: recipes/optick/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        tc.generate()
+
+        tc = CMakeDeps(self)
+        tc.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+
+    def build(self):
+        self._patch_sources()
```
Comment (uilianries) [inline recipes/optick/all/conanfile.py:69]:
```suggestion

    def build(self):
```

There are no dependencies to be installed, neither patches to be applied.

File: recipes/optick/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        if not is_msvc(self):
+            raise ConanInvalidConfiguration(f"{self.ref} can be built only by Visual Studio and msvc.")
+
+    def build_requirements(self):
+        pass
+
```
Comment (uilianries) [inline recipes/optick/all/conanfile.py:50]:
```suggestion
```

Not used

File: recipes/optick/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    def package(self):
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
```
Comment (uilianries) [inline recipes/optick/all/conanfile.py:83]:
```suggestion
        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
```

Cmake files are actually installed in lib/cmake:

```
optick/1.4.0.0: RUN: cmake --install "C:\J2\w\prod-v2\bsr\24240\ddadb\p\b\opticdf8109950c9bd\b\build" --config Release --prefix "C:/J2/w/prod-v2/bsr/24240/ddadb/p/b/opticdf8109950c9bd/p"
-- Installing: C:/J2/w/prod-v2/bsr/24240/ddadb/p/b/opticdf8109950c9bd/p/lib/OptickCore.lib
-- Installing: C:/J2/w/prod-v2/bsr/24240/ddadb/p/b/opticdf8109950c9bd/p/bin/OptickCore.dll
-- Installing: C:/J2/w/prod-v2/bsr/24240/ddadb/p/b/opticdf8109950c9bd/p/include/optick.h
-- Installing: C:/J2/w/prod-v2/bsr/24240/ddadb/p/b/opticdf8109950c9bd/p/include/optick.config.h
-- Installing: C:/J2/w/prod-v2/bsr/24240/ddadb/p/b/opticdf8109950c9bd/p/lib/cmake/Optick/OptickTargets.cmake
-- Installing: C:/J2/w/prod-v2/bsr/24240/ddadb/p/b/opticdf8109950c9bd/p/lib/cmake/Optick/OptickTargets-release.cmake
-- Installing: C:/J2/w/prod-v2/bsr/24240/ddadb/p/b/opticdf8109950c9bd/p/lib/cmake/Optick/OptickConfig.cmake
-- Installing: C:/J2/w/prod-v2/bsr/24240/ddadb/p/b/opticdf8109950c9bd/p/lib/cmake/Optick/OptickConfigVersion.cmake
```

File: recipes/optick/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import is_msvc_static_runtime, is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
```
Comment (uilianries) [inline recipes/optick/all/conanfile.py:6]:
```suggestion
from conan.tools.files import get, copy, rm, rmdir
from conan.tools.build import check_min_cppstd
from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
```

**Review by uilianries — COMMENTED** [review]
Thank you for your PR! Please, try to simplify your recipe a bit, running methods that are not used will only consume more CPU with no benefit.

**Comment by luizfeldmann** [comment]
Hi @uilianries,
My **v1** conan produces a _Find*.cmake_ and not a _*config.cmake_ which is being demanded by the _test_v1_package_.
I'm mostly familiar with **v2**, what am I missing here ?

File: recipes/optick/all/test_v1_package/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package"
```
Comment (uilianries) [inline recipes/optick/all/test_v1_package/conanfile.py:7]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```

- `cmake_find_package`: Will produce `Findxxxx.cmake`
- `cmake_find_package_multi`: Produces `xxxxConfig.cmake`

File: recipes/optick/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        pass
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        if not is_msvc(self):
+            raise ConanInvalidConfiguration(f"{self.ref} can be built only by Visual Studio and msvc.")
```
Comment (uilianries) [inline recipes/optick/all/conanfile.py:35]: @luizfeldmann Why is it not working for other platforms. Looking at https://github.com/bombomby/optick/blob/1.4.0.0/CMakeLists.txt I see no restrictions.

Comment (uilianries) [reply recipes/optick/all/conanfile.py:35]: > The profiler library, which is currently built in this recipe

Yes, and as far as I see, it generates for any platform:

- Windows:
```
optick/1.4.0.0: RUN: cmake --install "C:\J2\w\prod-v2\bsr@3\36200\eeccb\p\b\optic78d0377885876\b\build" --config Release --prefix "C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p"
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/OptickCore.lib
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/bin/OptickCore.dll
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/include/optick.h
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/include/optick.config.h
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/cmake/Optick/OptickTargets.cmake
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/cmake/Optick/OptickTargets-release.cmake
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/cmake/Optick/OptickConfig.cmake
-- Installing: C:/J2/w/prod-v2/bsr@3/36200/eeccb/p/b/optic78d0377885876/p/lib/cmake/Optick/OptickConfigVersion.cmake
```

- Mac
```
Install the project...
cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /tmp/install/lib/libOptickCore.dylib
-- Installing: /tmp/install/include/optick.h
-- Installing: /tmp/install/include/optick.config.h
-- Installing: /tmp/install/lib/cmake/Optick/OptickTargets.cmake
-- Installing: /tmp/install/lib/cmake/Optick/OptickTargets-release.cmake
-- Installing: /tmp/install/lib/cmake/Optick/OptickConfig.cmake
-- Installing: /tmp/install/lib/cmake/Optick/OptickConfigVersion.cmake
```

So it should not block other platforms, as the main objective is package libraries and headers.

> The GUI to receive the profiler data, which is a C# project and must build on Windows

How does it work? Is it a define in the header, a configuration file? Who generates it?

**Pull request closed without merging**